### ISSUE TITLE
Transform and Tell: Entity-Aware News Image Captioning

### ISSUE THREAD

**Issue by hkefka385** [hkefka385/paper_reading#50]
## 0. 論文 
Journal/Conference: **CVPR 2020** 
Title: **Transform and Tell: Entity-Aware News Image Captioning** 
Authors: Alasdair Tran, Alexander Mathews, Lexing Xie
URL: https://openaccess.thecvf.com/content_CVPR_2020/html/Tran_Transform_and_Tell_Entity-Aware_News_Image_Captioning_CVPR_2020_paper.html
## 1. どんなもの？ 
RoBERTaやTransformerや物体検出などを組み合わせて記事内の写真のキャプション生成モデルを提案
<img width="925" alt="スクリーンショット 2020-07-06 18 16 20" src="https://user-images.githubusercontent.com/10390578/86577191-d4987080-bfb4-11ea-885c-0271e75c0d23.png">

更に，それに適したデータセットも構築
## 2. 先行研究と比べてどこがすごい？
RoBERTaなどの導入 + 記事内用語の使用によっキャプションを生成しようとした点 
## 5. 議論はある？ 
ウェブページ、ソーシャルメディアの投稿、ユーザーコメントなど、他のタイプのリッチコンテキストを持つドメインの利用の可能性のある研究
Webページ、ソーシャルメディアフィード、医療文書などの文脈知識を持つ他の画像記述タスクへの発展性があるか？
## メモ
ウェブページ、ソーシャルメディアの投稿、ユーザーコメントなど、他のタイプのリッチコンテキストを持つドメインの利用の可能性のある研究
Webページ、ソーシャルメディアフィード、医療文書などの文脈知識を持つ他の画像記述タスクへの発展性があるか？

Abst
ニュース記事に埋め込まれた画像のキャプションを生成するタスク
2つの課題
・実世界の知識に依存 → Multi-head attentionなどを用いて単語と画像のオブジェクトを関連付ける
・言語的に豊か・流暢さ → byte-pair encodingを用いて単語部分のシーケンスを生成
GoodNewsデータセットに適用

1 Introduction
Captioningの弱点
・ 世界の知識の不足
・言語的表現力
→ キャプションの語彙とスタイルは時間の経過とともに変化
他のアプローチ︰テキスト抽出やテンプレート充填に依存

画像のオブジェクトと記事の単語にmulti-head attentionを計算し対処
未知の単語を生成するためにbyte-pair encodigを使用
Dynamic Convolutional Neural Networkを用いたモデル
New dataset NYKTimes800k 
言語的に豊かなキャプションを生成

貢献
・4つの異なるモダリティ (text, images, faces, objects)を考慮したattentionやtransformer, byte-pair encodingなどを取り入れたモデル
・新たなデータセットを作成
・複数の評価指標でパフォーマンスが向上

2 Related Works
以前のCaptioningの研究：RNN + CNN
Jeffrey Donahue, Lisa Anne Hendricks, Sergio Guadarrama,Marcus  Rohrbach,  Subhashini  Venugopalan,  Kate  Saenko,and Trevor Darrell.  Long-term recurrent convolutional net-works for visual recognition and description.  InThe IEEEConference  on  Computer  Vision  and  Pattern  Recognition(CVPR), June 2015
Oriol  Vinyals,  Alexander  Toshev,  Samy  Bengio,  and  Du-mitru Erhan.   Show and tell:  A neural image caption gen-erator.   InThe  IEEE  Conference  on  Computer  Vision  andPattern Recognition (CVPR), June 2015

どの画像領域にも注目しないAttention?
Jiasen Lu, Caiming Xiong, Devi Parikh, and Richard Socher.Knowing when to look: Adaptive attention via a visual sen-tinel for image captioning. InThe IEEE Conference on Com-puter Vision and Pattern Recognition (CVPR), July 2017
オブジェクト領域に特化
Weixuan Wang, Zhihong Chen, and Haifeng Hu.  Hierarchi-cal attention network for image captioning.  InProceedingsof the AAAI Conference on Artificial Intelligence, volume 33,pages 8957–8964, 201

ニュース記事のキャプション生成︰ただし新語は生成せず
Arnau  Ramisa,   Fei  Yan,   Francesc  Moreno-Noguer,   andKrystian Mikolajczyk. Breakingnews: Article annotation byimage and text processing.IEEE Transactions on PatternAnalysis and Machine Intelligence, 40:1072–1085, 2016.

制限1︰固定サイズの語彙
→テンプレートベースの方法︰
Ali  Furkan  Biten,  Lluis  Gomez,  Marcal  Rusinol,  and  Di-mosthenis Karatzas.  Good news, everyone!  context drivenentity-aware  captioning  for  news  images.InThe  IEEEConference  on  Computer  Vision  and  Pattern  Recognition(CVPR), June 201
★ナレッジグラフを用いてのEntityの検索：Di Lu, Spencer Whitehead, Lifu Huang, Heng Ji, and Shih-Fu Chang.  Entity-aware image caption generation.  InPro-ceedings of the 2018 Conference on Empirical Methods inNatural Language Processing, pages 4013–4023, Brussels,Belgium,  Oct.-Nov.  2018.  Association  for  ComputationalLinguistics
→本研究との違い：中間テンプレートを用いずにNamed Entityを持つキャプションを生成

Byte Pair Encodingを用いたNamed Entityの生成
★Sanqiang  Zhao,  Piyush  Sharma,  Tomer  Levinboim,  andRadu Soricut.   Informative image captioning with externalsources of information.  InProceedings of the 57th AnnualMeeting  of  the  Association  for  Computational  Linguistics,pages 6485–6494, Florence, Italy, July 2019. Association forComputational Linguistics
→Google Cloud Vision APiを用いて抽出したWeb entity labelをモデルの入力としているが、本研究ではニュース記事の中から関係するエンティティを抽出

3 The Transform and Tell Model
3.1 Encoders
・Image Encoders
ResNet-152で画像表現を抽出: 2048次元の49のベクトル → これをAttentionを用いて領域を注目

・Face encoders
Face baunding boxes検出のためのモデル：MTCNNの利用
Kaipeng Zhang, Zhanpeng Zhang, Zhifeng Li, and Yu Qiao.Joint face detection and alignment using multitask cascadedconvolutional  networks.IEEE  Signal  Processing  Letters,23:1499–1503, 2016
各顔のベクトル表現の取得：FaceNet
Florian Schroff,  Dmitry Kalenichenko,  and James Philbin.Facenet: A unified embedding for face recognition and clus-tering.   InThe  IEEE  Conference  on  Computer  Vision  andPattern Recognition (CVPR), June 2015
512次元のベクトルを取得

・Object Encoders
YoLOv3による物体検出 → ResNet-152でベクトル化

・Article Encoder
RoBERTaを用いて25 × 2048次元の表現を取得

3.2 Decoder
前のステップで生成されたtokenのembedding: z_0tやEncoderの出力を入力としてtransformer block内で計算
AdaptiveSoftmaxを用いてtokenを生成
＊adaptive softmax：高頻度グループの語+低頻度語のグループで通常のsoftmaxを行い，もし低頻度語のグループが選出された場合は低頻度語のグループ内でSoftmax計算
語の頻度分布の偏りに対応したsoftmax
5K、15K、30kの３つのクラスタに分割

Transformer block内部の動き
・Dynamic CNN
メモあり︰Felix Wu, Angela Fan, Alexei Baevski, Yann Dauphin, andMichael Auli.   Pay less attention with lightweight and dy-namic convolutions.  InInternational Conference on Learn-ing Representations, 2019
・Multi-head attention
keyとvalueが外部情報：Encoderによって生成された入力
query情報が内部情報：潜在状態

4 News Image Captioning
GoodNews
NYT800K
